src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -1,75 +1,120 @@
     @Getter
     private List<LogFile> logFiles;
-    private static Map<String, LogFile> fileHashToLogFile;
+    private final Map<String, LogFile> fileHashToLogFile;
     @Getter
     private final Pattern filePattern;
-    private final URI directoryURI;
-    private final Instant lastUpdated;
 
-    private LogFileGroup(List<LogFile> files, Pattern filePattern, URI directoryURI, Instant lastUpdated) {
+    private LogFileGroup(List<LogFile> files, Pattern filePattern, Map<String, LogFile> hashToFileIndex) {
         this.logFiles = files;
         this.filePattern = filePattern;
-        this.directoryURI = directoryURI;
-        this.lastUpdated = lastUpdated;
+        this.fileHashToLogFile = hashToFileIndex;
     }
 
     /**
      * Create a list of Logfiles that are sorted based on lastModified time.
-     * @param filePattern the fileNameRegex used for each component to recognize its log files.
+     *
+     * @param filePattern  the fileNameRegex used for each component to recognize its log files.
      * @param directoryURI the directory path of the log files of component.
-     * @param lastUpdated the saved updated time of the last uploaded log of a component.
+     * @param lastUpdated  the saved updated time of the last uploaded log of a component.
      * @return list of logFile.
      * @throws InvalidLogGroupException the exception if this is not a valid directory.
      */
-    public static LogFileGroup create(Pattern filePattern, URI directoryURI, Instant lastUpdated)
+    public static LogFileGroup create(Pattern filePattern, URI directoryURI, Instant lastUpdated) throws
+            InvalidLogGroupException {
+        return LogFileGroup.create(filePattern, directoryURI, lastUpdated,
+                Paths.get(DEFAULT_HARDLINKS_PATH));
+    }
+
+
+    /**
+     * Create a list of Logfiles that are sorted based on lastModified time.
+     *
+     * @param filePattern  the fileNameRegex used for each component to recognize its log files.
+     * @param directoryURI the directory path of the log files of component.
+     * @param lastUpdated  the saved updated time of the last uploaded log of a component.
+     * @param hardLinksDirectory path to store hardlinks
+     * @return list of logFile.
+     * @throws InvalidLogGroupException the exception if this is not a valid directory.
+     */
+    public static LogFileGroup create(Pattern filePattern, URI directoryURI, Instant lastUpdated,
+                                      Path hardLinksDirectory)
             throws InvalidLogGroupException {
         File folder = new File(directoryURI);
-        fileHashToLogFile = new ConcurrentHashMap<>();
 
         if (!folder.isDirectory()) {
             throw new InvalidLogGroupException(String.format("%s must be a directory", directoryURI));
         }
 
-        LogFile[] files = LogFile.of(folder.listFiles());
+        Path componentHardlinksDirectory = hardLinksDirectory.resolve(filePattern.toString());
```

Comment:
Is file pattern the right thing to use here as it  can be a regex? Also, why are we using a separate component hardlink directory for each component if we can do pattern matching there as well.

Reply:
It is not. That is why I mentioned the hard-link directory should be passed as a parameter but lets leave that for a separate pr because that would require test changes.

Using the file pattern here for now to make the hardlink dirs for each component unique for now.

The reason que are creating unique directories is because at the beginning we are deleting everything that is in that directory for simplicity and we wouldn't want to accidentally delete hard links for another component

Reply:
This makes having a separate component hardlink directory mandatory.

Reply:
Yeah, if we wanted to structure it that way. Alternatively we could look in a non unique hardlink directory for all the hardlinks that match the pattern and delete them before creating new ones